drivers/gpio/gpio_pcal6416a.c

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2023 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_pcal6416a
+
+#include <zephyr/logging/log.h>
+
+#include "gpio_pcal64xxa.h"
+
+LOG_MODULE_REGISTER(pcal6416a, CONFIG_GPIO_LOG_LEVEL);
+
+enum {
+	PCAL6416A_REG_INPUT_PORT_0 = 0x00,
+	PCAL6416A_REG_INPUT_PORT_1 = 0x01,
+	PCAL6416A_REG_OUTPUT_PORT_0 = 0x02,
+	PCAL6416A_REG_OUTPUT_PORT_1 = 0x03,
+	PCAL6416A_REG_POLARITY_INVERSION_0 = 0x04,
+	PCAL6416A_REG_POLARITY_INVERSION_1 = 0x05,
+	PCAL6416A_REG_CONFIGURATION_0 = 0x06,
+	PCAL6416A_REG_CONFIGURATION_1 = 0x07,
+	PCAL6416A_REG_OUTPUT_DRIVE_STRENGTH_0_0 = 0x40,
+	PCAL6416A_REG_OUTPUT_DRIVE_STRENGTH_0_1 = 0x41,
+	PCAL6416A_REG_OUTPUT_DRIVE_STRENGTH_1_0 = 0x42,
+	PCAL6416A_REG_OUTPUT_DRIVE_STRENGTH_1_1 = 0x43,
+	PCAL6416A_REG_INPUT_LATCH_0 = 0x44,
+	PCAL6416A_REG_INPUT_LATCH_1 = 0x45,
+	PCAL6416A_REG_PULL_UP_DOWN_ENABLE_0 = 0x46,
+	PCAL6416A_REG_PULL_UP_DOWN_ENABLE_1 = 0x47,
+	PCAL6416A_REG_PULL_UP_DOWN_SELECT_0 = 0x48,
+	PCAL6416A_REG_PULL_UP_DOWN_SELECT_1 = 0x49,
+	PCAL6416A_REG_INTERRUPT_MASK_0 = 0x4A,
+	PCAL6416A_REG_INTERRUPT_MASK_1 = 0x4B,
+	PCAL6416A_REG_INTERRUPT_STATUS_0 = 0x4C,
+	PCAL6416A_REG_INTERRUPT_STATUS_1 = 0x4D,
+	PCAL6416A_REG_OUTPUT_PORT_CONFIGURATION = 0x4F,
+};
+
+static int pcal6416a_i2c_write(const struct i2c_dt_spec *i2c, uint8_t register_address,
+			       uint8_t value)
+{
+	int rc;
+
+	LOG_DBG("writing to register 0x%02X value 0x%02X", register_address, value);
+	rc = i2c_reg_write_byte_dt(i2c, register_address, value);
+
+	if (rc != 0) {
+		LOG_ERR("unable to write to register 0x%02X, error %i", register_address, rc);
+	}
+
+	return rc;
+}
+
+static int pcal6416a_i2c_read(const struct i2c_dt_spec *i2c, uint8_t register_address,
+			      uint8_t *value)
+{
+	int rc;
+
+	rc = i2c_reg_read_byte_dt(i2c, register_address, value);
+	LOG_DBG("reading from register 0x%02X value 0x%02X", register_address, *value);
+
+	if (rc != 0) {
+		LOG_ERR("unable to read from register 0x%02X, error %i", register_address, rc);
+	}
+
+	return rc;
+}
+
+static int pcal6416a_pins_cfg_apply(const struct i2c_dt_spec *i2c,
```

Comment:
Please change to `pcal64xxa_pins_cfg_apply()`

Reply:
I do not consider this being a good idea, as it is actually different for PCAL6408A and PCAL6416A, due to the additional registers.